packages/bolt-connection/src/channel/deno/deno-channel.js

```diff
@@ -0,0 +1,331 @@
+/**
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable */
+import ChannelBuffer from '../channel-buf'
+import { newError, internal } from 'neo4j-driver-core'
+import { iterateReader } from 'https://deno.land/std@0.157.0/streams/conversion.ts';
+
+const {
+  util: { ENCRYPTION_OFF, ENCRYPTION_ON }
+} = internal
+
+let _CONNECTION_IDGEN = 0
+/**
+ * Create a new DenoChannel to be used in web browsers.
+ * @access private
+ */
+export default class DenoChannel {
+  /**
+   * Create new instance
+   * @param {ChannelConfig} config - configuration for this channel.
+   */
+  constructor (
+    config,
+    connect = _connect
+  ) {
+    this.id = _CONNECTION_IDGEN++
+    this._conn = null
+    this._pending = []
+    this._error = null
+    this._open = true
+    this._config = config
+
+    this._receiveTimeout = null
+    this._receiveTimeoutStarted = false
+    this._receiveTimeoutId = null
+
+    this._connectionErrorCode = config.connectionErrorCode
+    this._handleConnectionError = this._handleConnectionError.bind(this)
+    this._handleConnectionTerminated = this._handleConnectionTerminated.bind(
+      this
+    )
```

Comment:
Nit: if you re-order to

```suggestion
    this.id = _CONNECTION_IDGEN++
    this._conn = null
    this._pending = []
    this._open = true
    this._error = null
    this._handleConnectionError = this._handleConnectionError.bind(this)
    this._handleConnectionTerminated = this._handleConnectionTerminated.bind(
      this
    )
    this._connectionErrorCode = config.connectionErrorCode
    this._receiveTimeout = null
    this._receiveTimeoutStarted = false
    this._receiveTimeoutId = null

    this._config = config
```
it's much closer to `packages/bolt-connection/src/channel/node/node-channel.js` which might or might not make common fixes/refactoring easier later down the road (same for the method order).